celery/app/task.py

```diff
@@ -612,7 +612,7 @@ def signature_from_request(self, request=None, args=None, kwargs=None,
     subtask_from_request = signature_from_request  # XXX compat
 
     def retry(self, args=None, kwargs=None, exc=None, throw=True,
-              eta=None, countdown=None, max_retries=None, **options):
+              eta=None, countdown=None, max_retries=-1, **options):
```

Comment:
is max_retires=-1 actually needed?

Reply:
Needed cause "None" is used for "unlimited retries". Elsewhere each retry without "max_retries" set will be converted to "unlimited retries" indirectly